packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapController.m

```diff
@@ -250,15 +250,14 @@ - (void)onMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result {
     result(nil);
   } else if ([call.method isEqualToString:@"map#takeSnapshot"]) {
     if (self.mapView != nil) {
-      UIGraphicsImageRendererFormat *format = [UIGraphicsImageRendererFormat defaultFormat];
-      format.scale = [[UIScreen mainScreen] scale];
       UIGraphicsImageRenderer *renderer =
-          [[UIGraphicsImageRenderer alloc] initWithSize:self.mapView.frame.size format:format];
-
+          [[UIGraphicsImageRenderer alloc] initWithSize:self.mapView.bounds.size];
       UIImage *image = [renderer imageWithActions:^(UIGraphicsImageRendererContext *context) {
-        [self.mapView.layer renderInContext:context.CGContext];
+        [self.mapView drawViewHierarchyInRect:self.mapView.bounds afterScreenUpdates:YES];
```

Comment:
can you explain why this fixes the issue? also have you verified earlier ios version to make sure it still works?

Reply:
Hi @hellohuanlin ! This is actually suppose to do the same thing. The method i now use is alternative to the other one. I have made test on IOS 17 and also earlier version it look like it's working well :)

Reply:
i may miss something here - is it now doing different things (one fixes the issue and the other does not?) if that's the case, can you add some comments doucmenting this behavior?

Reply:
Can you add a comment there saying the other API originally used doesn't work in ios 17? (even if you don't know why it doesn't work).

Reply:
done

Reply:
Please don't put the whole chunk of code. How about this: 

> For some unknown reason `mapView.layer::renderInContext` API returns a blank image on iOS 17. So we have to use `drawViewHierarchyInRect` API.

Reply:
Let's use this ! :)